src/tap/grpc/server.rs

```diff
@@ -265,14 +266,47 @@ impl iface::Tap for Tap {
 
         let base_event = base_event(req, inspect);
 
+        let authority = inspect.authority(req).unwrap_or_default();
+
+        let headers = if req.version() == http::Version::HTTP_2 {
+            // If the request is HTTP/2, add the pseudo-header fields to the
+            // headers.
+            let pseudos = vec![
+                http_types::headers::Header {
+                    name: ":method".to_owned(),
+                    value: req.method().as_str().as_bytes().into(),
+                },
+                http_types::headers::Header {
+                    name: ":scheme".to_owned(),
+                    value: req
+                        .uri()
+                        .scheme_part()
+                        .map(|scheme| scheme.as_str().as_bytes().into())
+                        .unwrap_or_default(),
+                },
+                http_types::headers::Header {
+                    name: ":authority".to_owned(),
+                    value: authority.as_bytes().into(),
+                },
+                http_types::headers::Header {
+                    name: ":path".to_owned(),
+                    value: req.uri().path().as_bytes().into(),
```

Comment:
`Uri::path` doesn't perfectly match `:path`. The `:path` header includes the query as well, so you probably want `Uri::path_and_query`.